rust/agents/validator/src/server/eigen_node.rs

```diff
@@ -0,0 +1,272 @@
+use axum::{
+    http::StatusCode,
+    response::IntoResponse,
+    routing::{get, Router},
+    Json,
+};
+use derive_new::new;
+use hyperlane_base::CoreMetrics;
+use hyperlane_core::HyperlaneDomain;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+enum ServiceStatus {
+    Up,
+    Down,
+    Initializing,
+}
+
+// struct for body of /node
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+struct NodeInfo {
```

Comment:
If the structs are based on a spec, can you link the spec as a doc comment at the top of this file? (using `//!`)